### DISCUSSION TITLE
Camera autofocus when capturing input images for reconstruction

### DISCUSSION THREAD

**Discussion by nimafo** [colmap/colmap#1649] [Q&A]
Hi,
As you probably know, by using autofocus features on cameras, the intrinsic of a camera, i.e. focal length, changes, and using a single set of camera parameters [theoretically] should not work.

I would like to know what are the right program options when doing SfM in COLMAP in cases where autofocus is used. 
And in this case, is it possible/accurate/ practical to use the output calibration results of OpenCV as camera parameters in COLMAP?

**Comment by ahojnnes** [comment]
You should make sure to NOT share intrinisics in colmap. You could use opencv's calibration as an initialization for colmap but don't set the calibration to fixed in colmap.

**Reply by nimafo** [reply]
Thanks, @ahojnnes, this seems smart to use OpenCV data as initialization for COLMAP, can you give a hint on how could that be done?
